forge-ai/src/main/java/forge/ai/ComputerUtil.java

```diff
@@ -1165,42 +1177,244 @@ public static boolean castPermanentInMain1(final Player ai, final SpellAbility s
             }
         }
 
-        // cast Backup creatures in main 1 to pump attackers
-        if (cardState.hasKeyword(Keyword.BACKUP)) {
-            for (Card potentialAtkr: ai.getCreaturesInPlay()) {
-                if (ComputerUtilCard.doesCreatureAttackAI(ai, potentialAtkr)) {
+        // Triggered abilities
+        for (Trigger trigger : cardState.getTriggers()) {
+            final String triggerZones = trigger.getParam("TriggerZones");
+            if (!"Battlefield".equals(triggerZones)) continue;
```

Comment:
not all Trigger has TriggerZones set (unless you want to exclude ETB and LTB on purpose)

like for ChangedZone Trigger, it is set there, to the `getActiveZone` set:
https://github.com/Card-Forge/forge/blob/0abc224e1bfc7507c3f1e9d97872e9600629566b/forge-game/src/main/java/forge/game/trigger/TriggerChangesZone.java#L216